doc-content/shared-kie-docs/src/main/asciidoc/KieServer/controller-rest-api-con.adoc

```diff
@@ -0,0 +1,108 @@
+[id='controller-rest-api-con_{context}']
+= {CONTROLLER} REST API for {KIE_SERVER} templates and instances
+
+{PRODUCT} provides a {CONTROLLER} REST API that you can use to interact with your {KIE_SERVER} templates (configurations), {KIE_SERVER} instances (remote servers), and associated KIE containers (deployment units) in {PRODUCT} without using {CENTRAL}. This API support enables you to maintain your {PRODUCT} resources more efficiently and optimize your integration and development with {PRODUCT}.
+
+With the {CONTROLLER} REST API, you can perform the following actions:
+
+* Retrieve information about {KIE_SERVER} templates, instances, and associated KIE containers
+* Update, start, or stop KIE containers associated with {KIE_SERVER} templates and instances
+* Create, update, or delete {KIE_SERVER} templates
+* Create, update, or delete {KIE_SERVER} instances
+
+Requests to the {CONTROLLER} REST API require the following components:
+
+Authentication::
+The {CONTROLLER} REST API requires basic HTTP Authentication or token-based authentication for the following user roles, depending on controller type:
++
+--
+* `rest-all` user role if you installed {CENTRAL} and you want to use the built-in {CONTROLLER}
+* `kie-server` user role if you installed the {HEADLESS_CONTROLLER} separately from {CENTRAL}
+
+To view configured user roles for your {PRODUCT} distribution, navigate to `~/$SERVER_HOME/standalone/configuration/application-roles.properties` and `../application-users.properties`.
+
+To add a user with the `kie-server` role or the `rest-all` role or both, navigate to `~/$SERVER_HOME/bin` and run the following command with the role or roles specified:
+
+[source,bash]
+----
+$ ./add-user.sh -a --user <USERNAME> --password <PASSWORD> --role kie-server,rest-all
+----
+
+To configure the `kie-server` or `rest-all` user with {CONTROLLER} access, navigate to `~/$SERVER_HOME/standalone/configuration/standalone-full.xml`, uncomment the `org.kie.server` properties (if applicable), and add the controller user login credentials and controller location (if needed):
+
+[source,xml,subs="attributes+"]
+----
+<property name="org.kie.server.location" value="http://localhost:8080/kie-server/services/rest/server"/>
+<property name="org.kie.server.controller" value="http://localhost:8080/{URL_COMPONENT_CENTRAL}/rest/controller"/>
+<property name="org.kie.server.controller.user" value="baAdmin"/>
+<property name="org.kie.server.controller.pwd" value="password@1"/>
+<property name="org.kie.server.user" value="baAdmin"/>
```

Comment:
I would suggest removing _org.kie.server.user_ and _org.kie.server.pwd_ properties to avoid confusion. In here you want to show how the Kie Server is configured to connect to the remote controller only.

Reply:
@sterobin the reason I mentioned that is because this configuration only makes sense when you deploy both controller and kie-server in the same WildFly instance. If you're running it separately they would be used as: _org.kie.server.controller_ in Kie Server and _org.kie.server_ in the Controller instance.

Reply:
@cristianonicolai Done. I removed kie.server.user and kie.server.pwd